warehouse/templates/manage/publishing.html

```diff
@@ -0,0 +1,156 @@
+          {% if github_provider_form.owner.flags.required %}
+          <span class="form-group__required">{% trans %}(required){% endtrans %}</span>
+          {% endif %}
+        </label>
+        {{ github_provider_form.owner(placeholder=gettext("owner"), autocomplete="off", autocapitalize="off", spellcheck="false", class_="form-group__field", aria_describedby="owner-errors") }}
+        <div id="owner-errors">
+          {{ field_errors(github_provider_form.owner) }}
+        </div>
+      </div>
+      <div class="form-group">
+        <label for="repository" class="form-group__label">
+          {% trans %}Repository name{% endtrans %}
+          {% if github_provider_form.repository.flags.required %}
+          <span class="form-group__required">{% trans %}(required){% endtrans %}</span>
+          {% endif %}
+        </label>
+        {{ github_provider_form.repository(placeholder=gettext("repository"), autocomplete="off", autocapitalize="off", spellcheck="false", class_="form-group__field", **{"aria-describedby":"repository-errors"}) }}
+        <div id="repository-errors">
+          {{ field_errors(github_provider_form.repository) }}
+        </div>
+      </div>
+      <div class="form-group">
+        <label for="workflow_name" class="form-group__label">
+          {% trans %}Workflow name{% endtrans %}
+          {% if github_provider_form.workflow_name.flags.required %}
+          <span class="form-group__required">{% trans %}(required){% endtrans %}</span>
+          {% endif %}
+        </label>
+        {{ github_provider_form.workflow_name(placeholder=gettext("workflow.yml"), class_="form-group__field", autocomplete="off", **{"aria-describedby":"workflow_name-errors"}) }}
+        <div id="workflow_name-errors">
+          {{ field_errors(github_provider_form.workflow_name) }}
+        </div>
+      </div>
+      <div>
+        <input type="submit" value="{% trans %}Add{% endtrans %}" class="button button--primary">
+      </div>
+    </form>
+
+    <h2>{% trans %}Manage current providers{% endtrans %}</h2>
+    {% if project.oidc_providers %}
+    <table class="table">
+      <caption class="sr-only">
+        {% trans project_name=project.name %}OpenID Connect publishers associated with {{ project_name }}{% endtrans %}
+      </caption>
+      <thead>
+        <tr>
+          <th scope="col">{% trans %}Publisher{% endtrans %}</th>
+          <th scope="col">{% trans %}Specification{% endtrans %}</th>
+          <th></th>
+        </tr>
+      </thead>
+      <tbody>
+        {% for provider in project.oidc_providers %}
+        {{ provider_row(provider) }}
+        {% endfor %}
+      </tbody>
+    </table>
+    {% else %}
+    <p class="no-bottom-padding">{% trans %}No publishers are currently configured.{% endtrans %}</p>
+    {% endif %}
+    {% else %}
+    <p>
+      {% trans %}
+      OpenID Connect publishing isn't enabled yet! Come back soon.
+      {% endtrans %}
+    </p>
+    {% endif %}
```

Comment:
This shouldn't be necessary, this route should just 404 if the feature hasn't been enabled yet.

Reply:
Thanks; will delete.

Reply:
Done.